### DISCUSSION TITLE
Adguard is against dnsmasq and libvirt kvm is pro for dnsmasq and resolv.conf. How to reconcile them?

### DISCUSSION THREAD

**Discussion by tst32** [AdguardTeam/AdGuardHome#4539] [Q&A]
In order to run Adguard flawlessly, i disabled dnsmasq & systemd resolver so for the moment for 
```
sudo netstat -nulpd | grep 53
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           4506/avahi-daemon:  
udp6       0      0 :::53                   :::*                                6922/AdGuardHome    
udp6       0      0 :::5353                 :::*                                4506/avahi-daemon:  
 
```

```
sudo systemctl status systemd-resolved.service
● systemd-resolved.service - Network Name Resolution
     Loaded: loaded (/lib/systemd/system/systemd-resolved.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: man:systemd-resolved.service(8)
             https://www.freedesktop.org/wiki/Software/systemd/resolved
             https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers
             https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients


```
but when i run 
```
sudo virsh net-create tmp.xml
error: Failed to create network from tmp.xml
error: internal error: Child process (VIR_BRIDGE_NAME=newbr10 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/lib/libvirt/libvirt_leaseshelper) unexpected exit status 5: 
dnsmasq: directory /etc/resolv.conf for resolv-file is missing, cannot poll

```

if i do `sudo systemctl enable systemd-resolved.service` this will break adguard 
any suggestions?

**Comment by chiragvshroff** [comment]
> In order to run Adguard flawlessly, i disabled dnsmasq & systemd resolver so for the moment for 
> 
> ```
> 
> sudo netstat -nulpd | grep 53
> 
> udp        0      0 0.0.0.0:5353            0.0.0.0:*                           4506/avahi-daemon:  
> 
> udp6       0      0 :::53                   :::*                                6922/AdGuardHome    
> 
> udp6       0      0 :::5353                 :::*                                4506/avahi-daemon:  
> 
>  
> 
> ```
> 
> 
> 
> ```
> 
> sudo systemctl status systemd-resolved.service
> 
> ● systemd-resolved.service - Network Name Resolution
> 
>      Loaded: loaded (/lib/systemd/system/systemd-resolved.service; disabled; vendor preset: enabled)
> 
>      Active: inactive (dead)
> 
>        Docs: man:systemd-resolved.service(8)
> 
>              https://www.freedesktop.org/wiki/Software/systemd/resolved
> 
>              https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers
> 
>              https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients
> 
> 
> 
> 
> 
> ```
> 
> but when i run 
> 
> ```
> 
> sudo virsh net-create tmp.xml
> 
> error: Failed to create network from tmp.xml
> 
> error: internal error: Child process (VIR_BRIDGE_NAME=newbr10 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/lib/libvirt/libvirt_leaseshelper) unexpected exit status 5: 
> 
> dnsmasq: directory /etc/resolv.conf for resolv-file is missing, cannot poll
> 
> 
> 
> ```
> 
> 
> 
> if i do `sudo systemctl enable systemd-resolved.service` this will break adguard 
> 
> any suggestions?

Facing The Same issue did you find any solution?

**Comment by fernvenue** [comment]
They just wanna use the same port, you should use just one DNS service on your device, if you really wanna run multiple DNS service at the same time, just set different ports for them.
